util/src/main/java/tc/oc/pgm/util/menu/pattern/Dynamic5RowMenuArranger.java

```diff
@@ -0,0 +1,233 @@
+package tc.oc.pgm.util.menu.pattern;
+
+import java.util.List;
+import tc.oc.pgm.util.menu.InventoryMenuItem;
+import tc.oc.pgm.util.menu.InventoryMenuUtils;
+
+public class Dynamic5RowMenuArranger extends MenuArranger {
+
+  @Override
+  public List<InventoryMenuItem> arrangeItems(List<InventoryMenuItem> itemsWithoutSpaces) {
+
+    // Figure out which slots to use
+    int[] slots = new int[Math.min(itemsWithoutSpaces.size(), 18)];
+
+    switch (slots.length) {
```

Comment:
Is there a better way to do this? This it totally nebulous.

Reply:
It felt the most intuitive way to have a different display for each amount. I could always reduce the "customization" for less code?

Reply:
You can create a similar algorithm without hand-picked slots

Reply:
This could just be my opinion but the checkerboard pattern for player heads just seems strange. I don't see any reason for it as it just increases the number of pages required to display a full team due to the excess padding.

![image](https://user-images.githubusercontent.com/8608892/101390611-0478d680-38bb-11eb-88bb-a577acbd4591.png)

I think the teams should use some logic to spread them evenly but like you say I'm sure some math can be done for this.

![image](https://user-images.githubusercontent.com/8608892/101390584-fa56d800-38ba-11eb-92c9-9b37f95ad76f.png)

Reply:
Seems like the algorithm would just start in the upper right hand corner and skip a slot?